src/test/java/org/jenkinsci/plugins/workflow/cps/CpsScmFlowDefinitionTest.java

```diff
@@ -157,9 +163,24 @@
         p.setDefinition(def);
         WorkflowRun b = r.assertBuildStatusSuccess(p.scheduleBuild2(0));
         r.assertLogNotContains("Cloning the remote Git repository", b);
+        r.assertLogNotContains("Retrying after 10 seconds", b);
         r.assertLogContains("Obtained flow.groovy from git " + sampleRepo, b);
         r.assertLogContains("version one", b);
     }
+    
+    @Issue("JENKINS-39194")
+    @Test public void retry() throws Exception {
+        // We use an un-initialized repo here to test retry
+        WorkflowJob p = r.jenkins.createProject(WorkflowJob.class, "p");
+        GitStep step = new GitStep(invalidRepo.toString());
+        CpsScmFlowDefinition def = new CpsScmFlowDefinition(step.createSCM(), "flow.groovy");
+        def.setLightweight(false);
+        p.setDefinition(def);
+        r.jenkins.setScmCheckoutRetryCount(1);
+        WorkflowRun b = r.assertBuildStatus(Result.FAILURE, p.scheduleBuild2(0));
```

Comment:
Would be nice to extend this test to run a second build, this time using `r.waitForMessage("Retrying after 10 seconds", b)` and then initializing the repo and then asserting that the build does succeed with the second checkout.